networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -57,9 +60,11 @@ public class ConnectionManager extends Service {
   private final Counter failedConnectionCounter;
   private final PeerPools peerPools = new PeerPools();
   private final Collection<Predicate<DiscoveryPeer>> peerPredicates = new CopyOnWriteArrayList<>();
+  private final AtomicInteger startupCountdownStage = new AtomicInteger(5);
 
   private volatile long peerConnectedSubscriptionId;
   private volatile Cancellable periodicPeerSearch;
+  private volatile CountDownLatch activeSearch = new CountDownLatch(1);
```

Comment:
It would probably be better to have this be returned by `searchForPeers` and passed into `connectToBestPeers` rather than stored as a field. That way we know it's correctly tied to a specific search process and can't wind up being updated by multiple searches that may wind up executing at once (shouldn't happen but better to guarantee it and I think the code will wind up being easier to follow as well).

Reply:
The other approach which is more how we'd normally do it would be fore `searchForPeers` to return a future that only completes when the search process is complete.  That would let us avoid blocking a thread while waiting for the search process which would be better as the asyncRunner is typically shared and has a limited number of threads.

Reply:
Tried it with the `SafeFuture` return, looks better